File: IsoMarphicString.py
def isIsomorphic(s: str, t: str) -> bool:
    map_S = {}
    map_T = {}
    for i in range(len(s)):
        char_S = s[i]
        char_T = t[i]
        if char_S in map_S:

            if map_S[char_S] != char_T:
                return False

        else: 
            map_S[char_S] = char_T

        if char_T in map_T:

            if map_T[char_T] != char_S:
                return False

        else:
            map_T[char_T] = char_S
                
    return True
        
print(isIsomorphic("egg","add")) #Output = true

File: IsoMarphicString.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function (s, t) {
  let mapS = {},
    mapT = {};
  for (let i = 0; i < s.length; i++) {
    if (mapS[s[i]]) {
      if (mapS[s[i]] !== t[i]) return false;
    } else {
      mapS[s[i]] = t[i];
    }
    if (mapT[t[i]]) {
      if (mapT[t[i]] !== s[i]) return false;
    } else {
      mapT[t[i]] = s[i];
    }
  }
  return true;
};

console.log(isIsomorphic("egg", "add")); // Output: true
